bug_update.php

```diff
@@ -307,6 +307,11 @@
 	$t_new_custom_field_value = gpc_get_custom_field( 'custom_field_' . $t_cf_id, $t_cf_def['type'], '' );
 	$t_old_custom_field_value = custom_field_get_value( $t_cf_id, $f_bug_id );
 
+	# No need to update the field if its value has not changed
+	if( $t_old_custom_field_value == $t_new_custom_field_value ) {
```

Comment:
> so we need ===

Currently the logic does not use strict comparison (see https://github.com/mantisbt/mantisbt/blob/release-2.2.0/core/history_api.php#L78). 

I'm not saying this is correct, but if we change it here, I think we need to be consistent and update the history API as well.